### PR TITLE
MINOR: Add proper handling for Kafka Streams running applications

### PR DESCRIPTION
**Pull request by bbejeck**
This PR adds better handling for running a Kafka Streams application

### CODE REVIEW

**Review by davetroiano — APPROVED** [review]
couple of questions / whitespace suggestions - overall LGTM!

File: kafka-streams/src/main/java/io/confluent/developer/aggregate/StreamsAggregate.java [hunk 1]
```diff
@@ -33,21 +35,33 @@ public static void main(String[] args) throws IOException {
...
+                kafkaStreams.close(Duration.ofSeconds(2));
+                shutdownLatch.countDown();
+            }));
             TopicLoader.runProducer();
             kafkaStreams.start();
+            try {
+                shutdownLatch.await();
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
```
[the remark this answers is not in the document]
Comment (bbejeck) [reply kafka-streams/src/main/java/io/confluent/developer/aggregate/StreamsAggregate.java:63]: ack

File: kafka-streams/src/main/java/io/confluent/developer/basic/BasicStreams.java [hunk 2]
```diff
@@ -31,17 +34,29 @@ public static void main(String[] args) throws IOException {
...
-         try(KafkaStreams kafkaStreams = new KafkaStreams(builder.build(), streamsProps)) {
-             TopicLoader.runProducer();
-             kafkaStreams.start();
-         }
+        firstStream.peek((key, value) -> System.out.println("Incoming record - key " + key + " value " + value))
+                // filter records by making sure they contain the orderNumberStart variable from above HINT: use filter
+                // map the value to a new string by removing the orderNumberStart portion HINT: use mapValues
+                // only forward records where the value is 1000 or greater HINT: use filter and Long.parseLong
+                .peek((key, value) -> System.out.println("Outgoing record - key " + key + " value " + value));
+        //Write the results to an output topic defined above as outputTopic HINT: use "to" and Produced and Serdes.String()
```
[the remark this answers is not in the document]
Comment (bbejeck) [reply kafka-streams/src/main/java/io/confluent/developer/basic/BasicStreams.java:42]: ack

File: kafka-streams/src/main/java/io/confluent/developer/ktable/KTableExample.java [hunk 3]
```diff
@@ -33,20 +36,32 @@ public static void main(String[] args) throws IOException {
...
                 .mapValues(value -> value.substring(value.indexOf("-") + 1))
                 .filter((key, value) -> Long.parseLong(value) > 1000);
-                // Add a method here to covert the table to a stream
-                // Then uncomment the following two lines to view results on the console and write to a topic
-                //.peek((key, value) -> System.out.println("Outgoing record - key " +key +" value " + value))
-                //.to(outputTopic, Produced.with(Serdes.String(), Serdes.String()));
+        // Add a method here to covert the table to a stream
+        // Then uncomment the following two lines to view results on the console and write to a topic
+        //.peek((key, value) -> System.out.println("Outgoing record - key " +key +" value " + value))
+        //.to(outputTopic, Produced.with(Serdes.String(), Serdes.String()));
```
[the remark this answers is not in the document]
Comment (bbejeck) [reply kafka-streams/src/main/java/io/confluent/developer/ktable/KTableExample.java:47]: ack

**Comment by bbejeck** [comment]
@davetroiano updated per comments

**Pull request merged**